pkg/csi/service/vanilla/controller.go

```diff
@@ -269,18 +269,16 @@ func (c *controller) Init(config *cnsconfig.Config, version string) error {
 			for _, authMgr := range authMgrs {
 				go common.ComputeDatastoreMapForBlockVolumes(authMgr, config.Global.CSIAuthCheckIntervalInMin)
 			}
-			if !multivCenterTopologyDeployment {
-				for _, vcconfig := range c.managers.VcenterConfigs {
-					isvSANFileServicesSupported, err := c.managers.VcenterManager.IsvSANFileServicesSupported(ctx,
-						vcconfig.Host)
-					if err != nil {
-						return logger.LogNewErrorf(log, "failed to verify if vSAN file services is supported or not for vCenter: %s. "+
-							"Error:%+v", vcconfig.Host, err)
-					}
-					if isvSANFileServicesSupported {
-						for _, authMgr := range authMgrs {
-							go common.ComputeFSEnabledClustersToDsMap(authMgr, config.Global.CSIAuthCheckIntervalInMin)
-						}
+			for _, vcconfig := range c.managers.VcenterConfigs {
+				isvSANFileServicesSupported, err := c.managers.VcenterManager.IsvSANFileServicesSupported(ctx,
+					vcconfig.Host)
+				if err != nil {
+					return logger.LogNewErrorf(log, "failed to verify if vSAN file services is supported or not for vCenter: %s. "+
+						"Error:%+v", vcconfig.Host, err)
+				}
+				if isvSANFileServicesSupported {
```

Comment:
if one vc is older and does not support file service, and another is newer which support file service, we are starting go routine to compute fs enabled clusters for all VC in the loop.
can we fix this issue?

Reply:
Addressed in the latest change